src/js/models/tool.js

```diff
@@ -153,5 +153,11 @@ define(function (require, exports, module) {
      */
     Tool.prototype.toolOverlay = null;
 
+    /**
+     * Optional indicate is a tool is usable in vector make mode 
+     * @type {boolean}
+     */
+    Tool.prototype.handlevectorMaskMode = null;
```

Comment:
This is a tool specific flag, it shouldn't be declared in the tool model. The corresponding tools can just declare it themselves in their constructor and you can test for undefined where you need it.